### PR TITLE
feat: download all models from list and convert them to IR

### PR DESCRIPTION
**Pull request by vli11**
feat: download all models from list and convert them to IR

Closes: #242

## PR Checklist

- [x] Added label to the Pull Request for easier discoverability and search
- [x] Commit Message meets guidelines as indicated in the URL https://github.com/intel-retail/automated-self-checkout/blob/main/CONTRIBUTING.md
- [x] Every commit is a single defect fix and does not mix feature addition or changes
- [ ] Unit Tests have been added for new changes
- [ ] Updated Documentation as relevant to the changes
- [ ] All commented code has been removed
- [ ] If you've added a dependency, you've ensured license is compatible with repository license and clearly outlined the added dependency.
- [ ] PR change contains code related to security
- [ ] PR introduces changes that breaks compatibility with other modules (If YES, please provide details below)

## What are you changing?
adding dockerfile, makefile, script to download rest of ovms models and convert them to IR; all files created are under download_models directory

## Issue this PR will close

close: #242 

## Test Instructions if applicable
cd download_models
make build
make run
you should see output models downloaded in configs/opencv-ovms/models/2022/modelslist/

## If the there are associated PRs in other repositories, please link them here (i.e. intel-retail/automated-self-checkout )
download_models/models.lst has the full models list to download, it could take long time and big memory usage if you download it all, you can delete or comment out some to make smaller list for fast testing.

### CODE REVIEW

**Review by jim-wang-intel — CHANGES_REQUESTED** [review]
not sure want to download all models if not being used at all- some design discuss first please

**Comment by vli11** [comment]
> not sure want to download all models if not being used at all- some design discuss first please

that was the requirement from our board task, @brian-intel @antoniomtz should we download all or we can comment out some of the models from list?

**Comment by jim-wang-intel** [comment]
> > not sure want to download all models if not being used at all- some design discuss first please
> 
> that was the requirement from our board task, @brian-intel @antoniomtz should we download all or we can comment out some of the models from list?

yea, would like to discuss and i would like to challenge that requirements if no model being used at all - why download all?

File: download_models/Makefile [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# Copyright © 2023 Intel Corporation. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+.PHONY: build
+
+build:
+	docker build --tag download_models:dev -f Dockerfile.downloadModels .
+
+run:
+	docker run --net host --rm -v "$$PWD/../configs/opencv-ovms/models/2022:/2022" -e PROTOCOL_BUFFERS_PYTHON_IMPLEMENTATION=python download_models:dev
```
Comment (jim-wang-intel) [inline download_models/Makefile:10]: do we need host network here? 
```suggestion
	docker run --rm -v "$$PWD/../configs/opencv-ovms/models/2022:/2022" -e PROTOCOL_BUFFERS_PYTHON_IMPLEMENTATION=python download_models:dev
```

File: docs_src/OVMS/supportingDifferentModel.md [hunk 2]
```diff
@@ -13,4 +17,14 @@ OvmsClient:
...
+
+## Download Models
+You can download models by editing `download_models/models.lst` file, you can add new models(from https://github.com/openvinotoolkit/open_model_zoo/blob/master/demos/object_detection_demo/python/models.lst) to it or uncomment from existing list in this file, saved the file once editing is done. Then you can download the list using following steps:
+
+1. `cd download_models`
+2. `make build`
+3. `make run`
+
+after above steps, the downloaded models can be found in `configs/opencv-ovms/models/2022` directory.
+**_Note:_** model files in `configs/opencv-ovms/models/2022` directory will be replaced with new downloads if previously existed.
```
Comment (jim-wang-intel) [inline docs_src/OVMS/supportingDifferentModel.md:30]: how do i use this to download instance -segmentation and bit model? they are not covered in the list

**Review by jim-wang-intel — CHANGES_REQUESTED** [review]

File: docs_src/OVMS/supportingDifferentModel.md (hunk 2 above)
Comment (vli11) [reply docs_src/OVMS/supportingDifferentModel.md:30]: unfortunately open model zoo omz_downloader doesn't recognize these models, but luckily, we already have them in our code base. if our future OVMS models will be aligned with open model zoo's, then maybe we should look forward to open model zoo's models and just leave what we had.

File: download_models/Dockerfile.downloadModels
Comment (jim-wang-intel) [inline download_models/Dockerfile.downloadModels:1]: how big is the docker image? do we need everything from  https://github.com/openvinotoolkit/open_model_zoo.git?

**Review by jim-wang-intel — COMMENTED** [review]
so instead of asking user to do extra command: `make build / make run`, we could integrate both commands into our current model-downloader.sh.

**Comment by vli11** [comment]
> so instead of asking user to do extra command: `make build / make run`, we could integrate both commands into our current model-downloader.sh.

I think we can always improve it later once we are more confirming about generic pipeline flow.

**Review by jim-wang-intel — CHANGES_REQUESTED** [review]
[replace these](https://github.com/vli11/automated-self-checkout/blob/download_rest_models/download_models/downloadOVMSModels.sh#L83-L87) with the new model-list-downloader

**Comment by jim-wang-intel** [comment]
> > so instead of asking user to do extra command: `make build / make run`, we could integrate both commands into our current model-downloader.sh.
> 
> I think we can always improve it later once we are more confirming about generic pipeline flow.

it would be nice to see that integrates into the current process of benchmarking

**Comment by vli11** [comment]
> [replace these](https://github.com/vli11/automated-self-checkout/blob/download_rest_models/download_models/downloadOVMSModels.sh#L83-L87) with the new model-list-downloader



> [replace these](https://github.com/vli11/automated-self-checkout/blob/download_rest_models/download_models/downloadOVMSModels.sh#L83-L87) with the new model-list-downloader

we can do this clean up later once we are more confirming about generic pipeline flow.

File: download_models/downloadOVMSModels.sh [hunk 3]
```diff
@@ -82,8 +82,12 @@ fi
 
 if [ $personVehicleModelDownloaded -eq 0 ]; then
     echo "download people vehicle model..."
-    mkdir -p "$localPersonVehicleDetectionMdlDirName/FP16-INT8/1"
-    getOVMSModelFiles $personVehicleDetection $pipelineZooModel$personVehicleDetection $modelPrecisionFP16INT8 $localPersonVehicleDetectionMdlDirName
+    currentDir=$PWD
+    echo $currentDir
+    cd ../../../../download_models
```
Comment (jim-wang-intel) [inline download_models/downloadOVMSModels.sh:87]:
```suggestion
    cd  "$MODEL_EXEC_PATH"/../download_models"
```

File: download_models/downloadOVMSModels.sh [hunk 4]
```diff
@@ -82,8 +82,12 @@ fi
 
 if [ $personVehicleModelDownloaded -eq 0 ]; then
     echo "download people vehicle model..."
-    mkdir -p "$localPersonVehicleDetectionMdlDirName/FP16-INT8/1"
-    getOVMSModelFiles $personVehicleDetection $pipelineZooModel$personVehicleDetection $modelPrecisionFP16INT8 $localPersonVehicleDetectionMdlDirName
+    currentDir=$PWD
+    echo $currentDir
```
Comment (jim-wang-intel) [inline download_models/downloadOVMSModels.sh:86]: remove

File: download_models/downloadOVMSModels.sh [hunk 5]
```diff
@@ -82,8 +82,12 @@ fi
...
+    cd ../../../../download_models
+    make docker
+    make run
+    cd $currentDir
```
Comment (jim-wang-intel) [inline download_models/downloadOVMSModels.sh:90]:
```suggestion
    cd - 
```

**Review by jim-wang-intel — CHANGES_REQUESTED** [review]
fix reviewdog issues and suggested changes

File: download_models/downloadModelsFromList.sh [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+fi
+
+# create folder 1 under each precision FP directory to hold the .bin and .xml files
+cd intel
+for d in */ ; do
+    echo "$d"
+    cd "$d"
+    for FP_Dir in */ ; do
+        echo "$FP_Dir"
+        mkdir "$FP_Dir"1
```
Comment (jim-wang-intel) [inline download_models/downloadModelsFromList.sh:23]:
```suggestion
        mkdir -p "$FP_Dir"1
```

**Review by jim-wang-intel — APPROVED** [review]
LGTM, good job!

**Pull request merged**